script.js

```diff
@@ -24,4 +21,80 @@ function episodeCode(season, number) {
   let code = `S${s}E${num}`;
   return code;
 }
-window.onload = setup;
+
+const searchBox = document.getElementById("search");
+searchBox.addEventListener("input", handleSearchInput);
+function handleSearchInput(event) {
+  state.searchTerm = event.target.value;
+  console.log(state.searchTerm);
+  renderBySearch();
+}
+
+const episodeSelector = document.getElementById("episode-selector");
+episodeSelector.addEventListener("change", handleSelect);
+
+function handleSelect(event) {
+  state.selectedEpisodeId = event.target.value;
+  console.log(state.selectedEpisodeId);
+  renderBySelect();
+}
+
+function createEpisodeListItem(episode) {
+  const episodeListItem = document.getElementById("episode-list").content.cloneNode(true);
+  const option = episodeListItem.querySelector("option");
+  option.textContent = `${episodeCode(episode.season, episode.number)} - ${episode.name}`;
+  option.setAttribute("value", episode.id);
+
+  return episodeListItem;
+}
+
+function fillEpisodeList() {
+  const episodes = state.episodes;
+  for (e of episodes) {
+    const episodeListItem = createEpisodeListItem(e);
+    document.getElementById("episode-selector").append(episodeListItem);
+  }
+}
+
+function renderBySelect() {
+  searchBox.value = "";
+  state.searchTerm = "";
+  renderByFilter(filterBySelect);
+}
+
+function filterBySelect(episode) {
+  return state.selectedEpisodeId == episode.id;
```

Comment:
I would probaly inline this function - that is make it an arrow function and pass it to renderByFilter. It's only used in the one function above so ther's no need for it to be a proper funtion, but it's a matter of prefrence

Reply:
Inlined the function 👍